### ISSUE TITLE
 Local Minima와 Global Minimum에 대해 설명해주세요.

### ISSUE THREAD

**Issue by kimyuuuuuu** [AInsteins/AI-Tech-Interview#33]

**Comment by import-max** [comment]
- global minimum은 에러가 최소화되는 값
- local minima는 global minimum 을 찾는 과정 중 후보 minimum의 하나이다. gradient descent 에서 발생하는 문제 중 하나로, epoch가 돌면서 국소부위를 찾아 계속해서 기울기를 업데이트하게 된다. local minima가 global minimum인 것으로 판단되어 optimizer가 작동하는 경우, loss function 함수가 오작동된다.
- 이런 경우에 momentum , NAG, Adagrad, Adadelta, RMSprop, Adam을 통해 해결할 수 있다.

**Comment by kimyuuuuuu** [comment]
비용 함수(cost function)에서

- **Global Minimum**
    - 에러가 최소화되는 즉, 우리가 찾고자 하는 지점
- **Local Minima**
    - 에러가 최소가 될 수 있는 후보 지점 중 Global Minimum을 뺀 지점
        
        ⇒ 에러를 최소화시키는 최적의 파라미터를 찾는 문제에 있어서 파라미터 공간에 수많은 지역적인 홀(hole)들이 존재하여 이러한 local minima에 빠질 경우 전역적인 해(global minimum)를 찾기 힘들게 되는 문제를 일컫는다.
        
    - Local Minima는 자칫 에러가 **최소화되는 지점을 찾았다고 착각**할 수 있기에 **함정**에 비유할 수 있다.
        
        → 이를 해결하기 위해 Momentum과 같은 **최적화 알고리즘**을 사용하거나 **학습률(learning rate)를 잘 조절**하여 Local Minima에서 벗어날 수 있다.

**Comment by jarammm** [comment]
global minimum : 에러가 최소화되는 절대적인 지점
local minima : 에러가 최소가 될 수 있는 후보가 되는 지점(아래로 볼록한 지점) 중 Global Minimum을 제외한 지점들
Local Minima는 자칫 에러가 최소화되는 지점을 찾았다고 착각할 수 있기에 함정에 비유할 수 있다. 이를 해결하기 위해 Momentum과 같은 최적화 알고리즘을 사용하거나 학습률(learning rate)를 잘 조절하여 Local Minima에서 벗어날 수 있다.

**Comment by miinkang** [comment]
- 비용 함수(cost function)에서 나오는 개념
- global minimum
    - cost가 최소화되는 지점, 최종적으로 도달하고자 하는 지점
- local minimum
    - global minimum이 될 수 있는 후보들 중 global minimum을 뺀 지점들
    - local minimum을 global minimum으로 착각할 수 있다.
    - learning rate를 조절하거나, 방향성을 유지하는 momentum을 사용해 local minimum에서 벗어나도록 학습해야한다.
